### PR TITLE
IAP-4480: create elasticsearch page ...

### PR DESCRIPTION
**Pull request by kvedenskii-axway**
sections:
* intro
* prerequisites
* installation
* api manager configuration
* elasticsearch configuration
* manual data push
* recommended settings

Thank you for your contribution to the Axway-Open-Docs repo.

## Describe the changes

Enter a brief description of your changes to communicate to the reviewers what you changed and why.

## Deploy preview link

https://deploy-preview-2074--axway-open-docs.netlify.app/docs/apim_installation/apiportal_install/install_software_elastic/

## Checklist for contributors

Before submitting this PR, please make sure:

* [ ] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [ ] You have signed the [Axway CLA](https://cla.axway.com/)
* [ ] You have verified the technical accuracy of your change
* [ ] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [ ] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)
* [ ] You have verified that all status checks have passed

_Put an x in the boxes that apply. This is simply a reminder of what we are going to look for before merging your change._

### CODE REVIEW

**Comment by andreamussap** [comment]
> Software installation

@dginevaAxway it's important to add the information here that this is only available for Software installation, and not in docker containers. But, there's not need to add a note in the docker part of the docs. Thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+API Portal now can store Apis and Applications from API Manager into your Elasticsearch server and then fetch them from it instead of API Manager.
+This approach is very useful for use cases with a lot of APIs and Applications
+It provides benefits like:
+
+* Having pagination on API Catalog page and Applications page
+* Configure how many items you want to see per page
+* Much better time and memory performance for API Catalog and Applications page
+
+{{< alert title="Note" color="primary" >}}Once you enable and configure Elasticsearch the API Portal Applications and APIs list pages will be loaded with data from your Elasticsearch instance instead of API Manager. Any change on these resources done in the API Manager will be reflected in the API Portal when the next scheduled indexation is executed (see _step 4_ and  _step 5_ in [Elasticsearch configuration in API Portal](#elasticsearch-configuration-in-api-portal) in this page) or after triggering the indexaion manually see ([Push data to Elasticsearch manually](#push-data-to-elasticsearch-manually) setion on this page) {{< /alert >}}
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:17]: There's no need to embed this in a Note. It's just part of the Description/explanation. Use the Note admonition to add the information about being available only for software installation, not in containers

Comment (varlogerr) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md]: Fixed, thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+  "title": "Using Elasticsearch with API Portal",
+  "linkTitle": "Install Elasticsearch",
+  "weight": "50",
+  "date": "2021-08-23",
+  "description": "Optionally install Elasticsearch for better performance."
```
[the remark this answers is not in the document]
Comment (varlogerr) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md]: Fixed, thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{
+  "title": "Using Elasticsearch with API Portal",
```
[the remark this answers is not in the document]
Comment (kvedenskii-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md]: Fixed, thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kvedenskii-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md]: Fixed, thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 1 above)
Comment (kvedenskii-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md]: Fixed, thanks

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{
+  "title": "Configure Elasticsearch in API Portal",
+  "linkTitle": "Install Elasticsearch",
+  "weight": "50",
+  "date": "2021-08-23",
+  "description": "Configure Elasticsearch in your API Portal to improve API and Applications management performance."
+}
+
+API Portal now can store Apis and Applications from API Manager into your Elasticsearch server and then fetch them from it instead of API Manager.
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:9]: as per @dginevaAxway comments, it seems that this page is still missing information about ElasticSearch bein available only for Software installation ?

Comment (kvedenskii-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:9]: Fixed this by a warning note under "Elasticsearch configuration in API Portal" section

**Comment by kvedenskii-axway** [comment]
> Can you rephrase this "Cron scheduling will not work in API Portal docker container" to "Cron scheduling is not yet available when using API Portal in a docker container"

Done

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+{
+  "title": "Configure Elasticsearch",
+  "linkTitle": "Configure Elasticsearch",
+  "weight": "50",
+  "date": "2021-08-23",
+  "description": "Configure Elasticsearch in your API Portal to improve API and Applications management performance."
+}
+
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:8]: what exactly improve the performance? Just connecting APIP with Elasticsearch, or the fact that you can create pagination?

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+When you connect your API Portal to an Elasticsearch server, API Portal can fetch all the API and applications data from API Manager instances and load it into the Elasticsearch datastore.
+
+You can also configure how many items per page you wish to see in the API catalog and Applications pages, which improves performance when there is a big number of APIs and Applications because only the configured number of resources per page are fetched and loaded into the server memory instead of the full list.
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:11]: you don't say anywhere in the page how to do this

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## Prerequisites
+
+* You must have an up and running Elasticsearch server.
+* You must have `cron` and `crontab` installed and `crond` service running in your API Portal server. (You can usually get `cron` by default with RHEL and CentOS Linux.)
+* Connecting Elasticsearch to a single node in API Portal is enough.
+
+## Elasticsearch security practices
+
+We recommend setting the following in your Elasticsearch server:
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:21]: 'When setting up your Elasticsearch server, observe the following:' -> would this be better?

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+   * Protocol
+   * Host
+   * Port
+   * Username
+   * User Password
+4. Click **Save**.
+
+## Configure a schedule to push data to Elasticsearch
+
+After connecting API to Elasticsearch, populate the crontab fields to create a schedule to push data to Elasticsearch.
```
Comment (andreamussap) [inline content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:54]: (to push data to Elasticsearch) -> to automatically push data to Elasticsearch

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 5 above)
Comment (astoyanovAxway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:8]: It is explained bellow what improves the performance: 
"....which improves performance when there is a big number of APIs and Applications because only the configured number of resources per page are fetched and loaded into the server memory instead of the full list."

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 6 above)
Comment (astoyanovAxway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:11]: "You can also configure how many items per page you wish to see in the API catalog and Applications"
This is actually feature of the pagination itself that's why I thought it is intuitive enough without explaining it further

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 8 above)
Comment (astoyanovAxway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:54]: agree

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 5 above)
Comment (andreamussap) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:8]: so I'm assuming that when you Configure Elasticsearch in your API Portal, this allows you to 
'_configure how many items per page you wish to see in the API catalog and Applications pages_' and this is what improves performance

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 6 above)
Comment (andreamussap) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:11]: This page is focused on the UI part of the application, so we tell the user how to perform tasks by clicking here or there and configure things. You say that the user can configure something, but doesn't say how. I wouldn't know how to start it, but I'm not APIP user. If you think that this is intuitive enough that any user will know how to do it, then it's ok. 
cc @jowens-axway

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 7 above)
Comment (jowens-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:21]: 'When setting up your Elasticsearch server, observe the following:' = this is better, please use

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 5 above)
Comment (jowens-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:8]: yes, basically if the default results per page is 10, it only fetches 10 instead of all of them. there is a performance gain there straightaway.

File: content/en/docs/apim_installation/apiportal_install/install_software_elastic.md (hunk 6 above)
Comment (jowens-axway) [reply content/en/docs/apim_installation/apiportal_install/install_software_elastic.md:11]: "how to" connect portal to elastic seems to be covered in the section below "## Connect API Portal with Elasticsearch"
